cl/corpus_importer/management/commands/harvard_merge.py

```diff
@@ -0,0 +1,613 @@
+        if len(cluster_case_name) > len(cluster_case_name_full):
+            # Swap field values
+            update_dict["case_name"] = cluster_case_name_full
+            update_dict["case_name_full"] = cluster_case_name
+
+    if update_dict:
+        OpinionCluster.objects.filter(id=cluster_id).update(**update_dict)
+
+
+def merge_date_filed(cluster_id: int, harvard_data: dict) -> None:
+    """Merge date filed
+
+    :param cluster_id: The cluster id of the merging item
+    :param harvard_data: json data from harvard case
+    :return: None
+    """
+    cluster = OpinionCluster.objects.get(id=cluster_id)
+    harvard_date_filed = harvard_data.get("decision_date")
+    cluster_date_filed = cluster.date_filed
+    merge_cluster_dates(
+        cluster_id, "date_filed", (harvard_date_filed, cluster_date_filed)
+    )
+
+
+def merge_overlapping_data(cluster_id: int, clean_dictionary: dict) -> None:
+    """Merge overlapping data
+
+    :param cluster_id: the cluster id
+    :param clean_dictionary: the dictionary of data to merge
+    :return: None
+    """
+    if clean_dictionary != {}:
+        logging.info(f"Merging complete for: {cluster_id}")
+        return
+
+    long_fields = [
+        "syllabus",
+        "summary",
+        "history",
+        "headnotes",
+        "correction",
+        "cross_reference",
+        "disposition",
+    ]
+
+    for field_name in clean_dictionary.keys():
+        if field_name in long_fields:
+            merge_long_fields(
+                cluster_id,
+                field_name,
+                clean_dictionary.get(field_name),
+            )
+        elif field_name in ["other_dates"]:
+            merge_cluster_dates(
+                cluster_id,
+                field_name,
+                clean_dictionary.get(field_name),
+            )
+        elif field_name == "judges":
+            merge_judges(
+                cluster_id,
+                clean_dictionary.get(field_name),
+            )
+        elif field_name == "attorneys":
+            merge_strings(
+                cluster_id,
+                field_name,
+                clean_dictionary.get(field_name, ""),
+            )
+        else:
+            logging.info(f"Field not considered in the process: {field_name}")
+
+
+def update_docket_source(cluster_id: int) -> None:
+    """Update docket source and complete
+
+    :param cluster_id: the cluster id
+    :return: None
+    """
+    docket = OpinionCluster.objects.get(id=cluster_id).docket
+    source = docket.source
+    docket.source = Docket.HARVARD + source
+    docket.save()
+
+
+def update_cluster_source(cluster_id: int) -> None:
+    """Update cluster source
+
+    :param cluster_id: cluster id to update
+    :return: None
+    """
+    cluster = OpinionCluster.objects.get(id=cluster_id)
+    if cluster.source == "C":
+        # court website source updated to court website merged with Harvard
+        cluster.source = "CU"
+        cluster.save()
+
```

Comment:
this method is too narrow

Take for example 
`https://www.courtlistener.com/opinion/2616280/something-something/`

here this is an opinion that needs to be merged - and it's source is L.  This method would change the value from 

Lawbox - to Harvard and Court website.  

We already check at the start of the method if the Harvard Is part of the source.  We need to add the source to check against one offs at the start and then we simply need to update the source by adding U and matching it.

in the example above `LU` 

Which unfortunately is not listed as an option.

Reply:
Also- we need to update docket, and we need to probably sync the two sources.